### PR TITLE
Fix for CN-4

### PR DESCRIPTION
**Pull request by irfanurrehman**
This PR is based on top of #366.
Please ensure that #366 is merged before this.
@ading1977 @evat-pm

### CODE REVIEW

File: pkg/discovery/worker/group_metrics_collector.go [hunk 1]
```diff
@@ -115,3 +114,11 @@ func (collector *GroupMetricsCollector) getGroupName(etype metrics.DiscoveredEnt
 
 	return ownerTypeString, ownerString, nil
 }
+
+func groupKey(kind, ns, name string) string {
```
Comment (ading1977) [inline pkg/discovery/worker/group_metrics_collector.go]: I could be wrong, but is this function being used anywhere?

Comment (pallavidn) [reply pkg/discovery/worker/group_metrics_collector.go]: same question

Comment (irfanurrehman) [reply pkg/discovery/worker/group_metrics_collector.go]: Apologies, I did intend to remove it but I published this PR probably half asleep late last night.. :)
Updated!

**Comment by irfanurrehman** [comment]
> Can you confirm if the main change here to include the namespace in the group name ? The rest of the code is the restructuring and the fix that was already done in the PR#366.

@pallavidn yes, as listed in the title comment of this PR, this is on top of #366 so only the last commit in this PR is reviewable here.

**Pull request merged**